src/Serilog.Sinks.MSSqlServer/Sinks/MSSqlServer/MSSqlServerSink.cs

```diff
@@ -332,8 +330,21 @@ private string LogEventToJson(LogEvent logEvent)
                 var columnName = property.Key;
                 var columnType = row.Table.Columns[columnName].DataType;
                 object conversion;
+
                 var scalarValue = property.Value as ScalarValue;
-                if (scalarValue != null && TryChangeType(scalarValue.Value, columnType, out conversion))
+                if (scalarValue == null)
+                {
+                    row[columnName] = property.Value.ToString();
```

Comment:
The "row[columnName] = property.Value.ToString();" line would have been executed prior to this change if scalarValue was null.
I kept it there as is since I did not want to change existing behavior more than I needed to in order to fix the issue.

I don't know what the exact intentions of that line is, but my guess is that if it isn't a ScalarValue (and it's one of DictionaryValue, SequenceValue, StructureValue, etc...), that those can't be converted to a particular data type using TryChangeType.
ToString() in that cause presumably would provide a string representation of those non-scalar types.